### PR TITLE
Lazy `AllPackages` (2)

### PR DESCRIPTION
**Pull request by andrewbranch**
#764 + fixes in the last couple commits

### CODE REVIEW

File: packages/publisher/src/generate-packages.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ if (require.main === module) {
   const tgz = !!yargs.argv.tgz;
   logUncaughtErrors(async () => {
     const log = loggerWithErrors()[0];
-    const options = { ...defaultLocalOptions, definitelyTypedPath: outputDirPath };
+    const options = { ...defaultLocalOptions };
```
Comment (jakebailey) [inline packages/publisher/src/generate-packages.ts:41]: Hm, how did this work before? I noticed this before when debugging publishing the first time and wasn't sure why we were setting this to an output path...

Comment (andrewbranch) [reply packages/publisher/src/generate-packages.ts:41]: No idea! I don’t think I’ve ever run `pnpm generate` before but it was definitely broken for me

**Review by jakebailey — APPROVED** [review]

File: packages/definitions-parser/src/lib/definition-parser.ts [hunk 2]
```diff
@@ -254,7 +254,7 @@ export function getFiles(
   moduleResolutionHost: ts.ModuleResolutionHost
 ): readonly FilesForSingleTypeScriptVersion[] {
   const errors = [];
-  const rootDir = dt.subDir("types").subDir(typingsData.typesDirectoryName);
+  const rootDir = dt.subDir("types").subDir(typingsData.subDirectoryPath);
```
Comment (sandersn) [inline packages/definitions-parser/src/lib/definition-parser.ts:257]: this is the version that includes /v15, eg `react/v15`, right? `rootDir` sounds like a misleading name for it.

**Review by sandersn — APPROVED** [review]

**Comment by andrewbranch** [comment]
I’ll let you guys decide if/when to merge this, as I may not have the chance to watch it closely today. I tested generating packages locally so I think it’s good, but maybe you’d prefer to merge it Monday 🤷

**Comment by sandersn** [comment]
I'd prefer Monday, or Wednesday when both Jake and I will have a chance to watch it.

**Comment by jakebailey** [comment]
Fixed merge conflicts and also removed the references to `parse` from the CI jobs, which weren't updated. This also fixes #780 so probably want to try and merge tomorrow morning.

**Review by sandersn — APPROVED** [review]

**Pull request merged**